bindings/graphql/graphql.go

```diff
@@ -0,0 +1,172 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package graphql
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+	graphql "github.com/machinebox/graphql"
+)
+
+const (
+	// configurations to connect to GraphQL
+	connectionEndPointKey = "endpoint"
+
+	// keys from request's metadata
+	commandQuery    = "query"
+	commandMutation = "mutation"
+
+	// keys from response's metadata
+	respOpKey        = "operation"
+	respStartTimeKey = "start-time"
+	respEndTimeKey   = "end-time"
+	respDurationKey  = "duration"
+)
+
+// GraphQL represents GraphQL output bindings
+type GraphQL struct {
+	client *graphql.Client
+	header map[string]string
+	logger logger.Logger
+}
+
+var _ = bindings.OutputBinding(&GraphQL{})
+
+// NewGraphQL returns a new GraphQL binding instance
+func NewGraphQL(logger logger.Logger) *GraphQL {
+	var gql GraphQL
```

Comment:
Simplification:

```golang
return &GraphQL{
    logger: logger
}
```

Reply:
@pkedy I created that way to avoid getting linting warning `graphql.go:50:10  exhaustivestruct  client, header are missing in GraphQL'